decorate.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError 'Subclasses must implement correct_name method'
+  end
+end
+
+class Decorate < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorate
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorate
+  def correct_name
+    @nameable.correct_name[0...10] if @nameable.correct_name.length > 10
+  end
+end
```

Comment:
- [ ] I would suggest you move the person as per requirements on  this file since all of the required classes are here and it would be much easier to test the results and it would not run on errors also The class Person is defined after it is being used, so please update this file to something similar to this:
```ruby
require_relative 'person'

class Person < Nameable
  attr_accessor :name, :age

  def initialize(name = 'Unknown', age = nil)
    @name = name
    @age = age
  end

  def correct_name
    name
  end
end

```
With this change the following test cases will pass and you'll meet the requiremnets:
```ruby

person = Person.new('maximilianus', 22)
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name
```

Reply:
Thank you for the review. But I am a bit confused, are you suggesting moving all of today's work to the person file?

Reply:
- [ ] No leave everything as it is on the person.rb file just on the decorate.rb file add the require relative and the other part of Person where you inherit from nameable